### ISSUE TITLE
Resolve multiple redirects are created with the same "from" pathname build errors

### ISSUE THREAD

**Issue by alfarok** [w3f/polkadot-wiki#3448]
These errors can be seen when looking through the Polkadot or Kusama Staging build and deploy jobs.  The error is thrown several times stating **It is not possible to redirect the same pathname to multiple destinations**.

Console output of a single instance of the error:
```
Error:  @docusaurus/plugin-client-redirects: multiple redirects are created with the same "from" pathname: "/docs/en/build-build-with-polkadot"
It is not possible to redirect the same pathname to multiple destinations: 
- {"from":"/docs/en/build-build-with-polkadot","to":"/docs/build-build-with-polkadot"}
- {"from":"/docs/en/build-build-with-polkadot","to":"/docs/build-build-with-polkadot"}
```

**Comment by alfarok** [comment]
Full log of conflicts:
```
[INFO] Duplicate routes found!
- Attempting to create page at /docs/build-build-with-polkadot, but a page already exists at this route.
- Attempting to create page at /docs/build-hackathon, but a page already exists at this route.
- Attempting to create page at /docs/build-oracle, but a page already exists at this route.
- Attempting to create page at /docs/build-pdk, but a page already exists at this route.
- Attempting to create page at /docs/build-smart-contracts, but a page already exists at this route.
- Attempting to create page at /docs/build-storage, but a page already exists at this route.
- Attempting to create page at /docs/build-tools-index, but a page already exists at this route.
- Attempting to create page at /docs/build-wallets, but a page already exists at this route.
- Attempting to create page at /docs/ens, but a page already exists at this route.
- Attempting to create page at /docs/learn-auction, but a page already exists at this route.
- Attempting to create page at /docs/learn-balance-transfers, but a page already exists at this route.
- Attempting to create page at /docs/learn-crowdloans, but a page already exists at this route.
- Attempting to create page at /docs/learn-governance, but a page already exists at this route.
- Attempting to create page at /docs/learn-identity, but a page already exists at this route.
- Attempting to create page at /docs/learn-parachains, but a page already exists at this route.
- Attempting to create page at /docs/learn-parathreads, but a page already exists at this route.
- Attempting to create page at /docs/learn-registrar, but a page already exists at this route.
- Attempting to create page at /docs/learn-treasury, but a page already exists at this route.
- Attempting to create page at /docs/maintain-errors, but a page already exists at this route.
- Attempting to create page at /docs/maintain-guides-how-to-nominate-kusama, but a page already exists at this route.
- Attempting to create page at /docs/maintain-guides-how-to-stop-validating, but a page already exists at this route.
- Attempting to create page at /docs/maintain-guides-how-to-validate-kusama, but a page already exists at this route.
- Attempting to create page at /docs/maintain-guides-validator-community, but a page already exists at this route.
- Attempting to create page at /docs/thousand-validators, but a page already exists at this route.
This could lead to non-deterministic routing behavior.
```

**Comment by DrW3RK** [comment]
These may be two different issues (Duplicate routes vs multiple redirects).

The redirects are a result of this code snippet in docusuarus config file https://github.com/w3f/polkadot-wiki/blob/1e5f0a4d92d3c6398c72c9d462b5ef17073198cf/polkadot-wiki/docusaurus.config.js#L72

```
plugins: [
    'remark-docusaurus-tabs',
    [
      '@docusaurus/plugin-client-redirects',
      {
        redirects: [
          {
            to: '/',

            from: ['/en/latest', '/en/'],
          },
        ],
        createRedirects: function (existingPath) {
          if (existingPath.startsWith('/docs/')) {
            return [existingPath.replace('/docs/', '/docs/en/')];
          }
        },
      },
    ],
  ],

```

**Comment by alfarok** [comment]
I looked into this a little yesterday afternoon.  Like you said above there are 2 sets of errors being generated.  I think the routes are auto-generated with duplicates (which throws the first error) then when we hit the redirect code above and try to create redirects for all the same routes (where the second error occurs).  I need to look more in-depth at how the routes are being generated.  I see a `routes.js` file that gets generated in the `.docusaurus` folder for each project and this file contains the dupes.

**Comment by alfarok** [comment]
@DrW3RK I think I finally figured this out!  The duplicate routes are coming from `mirror` pages.  I am not familiar with the functionality of these yet but it seem like they are files that are auto-generated from existing markdown files and prefixed with `mirror-`.

There are 24 errors mentioned above from the console.  There are also 24 files that get auto-generated with the prefix `mirror` that match all the file names with duplicate routes.  When a mirror file is created the slug for the mirror and original page are the same so this is the first duplicate error that fires.  Then another error surfaces for the same issue when the redirect hits.

I need to learn more about what these `mirror-` pages are doing to understand the solution here.

**Comment by alfarok** [comment]
Example:

![image](https://user-images.githubusercontent.com/13341935/177639952-ff4b3e5c-dd48-40a1-958e-8379d44ab5e6.png)

**Comment by alfarok** [comment]
After reading [this](https://github.com/w3f/polkadot-wiki#mirror-pages) I think I understand why these pages are created.  One option for removing the errors could be to append the same prefix to the slug so it creates a new route for the mirrored pages, unless Docusaurus has resolved this in a newer release?

Since the original and the mirror share the same slug which ever page is compiled last will be what content is displayed at that route.

**Comment by alfarok** [comment]
Resolved with #3512
